### PR TITLE
Fix `LexicographicalAnnotationAttributeListing` string sorting

### PR DESCRIPTION
**Pull request by Ptijohn**
Suggested commit message:
```
Fix sorting of annotation nodes (#281)

By ignoring leading and trailing escaped double quotes.
```

### CODE REVIEW

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/LexicographicalAnnotationAttributeListing.java [hunk 1]
```diff
@@ -203,8 +203,10 @@ private ImmutableList<String> tokenize(Tree node) {
         /*
          * Tokens are split on `=` so that e.g. inline Spring property declarations are properly
          * sorted by key, then value.
+         * Leading and trailing double quotes are ignored for sorting purposes.
          */
-        return ImmutableList.copyOf(SourceCode.treeToString(node, state).split("=", -1));
+        return ImmutableList.copyOf(
+            SourceCode.treeToString(node, state).replaceAll("^\"|\"$", "").split("=", -1));
```
Comment (Ptijohn) [inline error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/LexicographicalAnnotationAttributeListing.java:209]: Not sure if the right fix tbh.
But after loads of testing, I finally got that the escaped quotes were considered in the sorting, which we actually don't want, and we want to sort what's inside the string. 
Thus filtering out the leading and trailing quotes (if exists) from there.

Comment (Stephan202) [reply error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/LexicographicalAnnotationAttributeListing.java]: Thanks for tracking down the source of the odd sorting! It looks like we should use `ASTHelpers#constValue` here; I'll push a proposal :)

**Review by Stephan202 — APPROVED** [review]
Nice!

Alternative suggested commit message:
```
Fix `LexicographicalAnnotationAttributeListing` string sorting (#281)

In particular, the empty string is now ordered first.
```

**Comment by Stephan202** [comment]
Had a shower thought; added one more commit.

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/LexicographicalAnnotationAttributeListing.java (hunk 1 above)
Comment (Ptijohn) [reply error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/LexicographicalAnnotationAttributeListing.java]: Oh, didn't know about this one :D 
Thanks @Stephan202 for pushing those extras!

**Review by rickie — APPROVED** [review]
Thanks for picking this up @Ptijohn ! It's nice to have this fixed after quite a while 😄.

**Pull request merged**